### PR TITLE
Prevent bad/failing stringstream to crash the export progress

### PR DESCRIPTION
**Pull request by angelowang**
assimp source is using Tabs and it's quite hard to type Tabs in a space-only VS.

### CODE REVIEW

File: code/ColladaExporter.cpp [hunk 1]
```diff
@@ -81,6 +81,10 @@ void ExportSceneCollada(const char* pFile, IOSystem* pIOSystem, const aiScene* p
 	// invoke the exporter 
 	ColladaExporter iDoTheExportThing( pScene, pIOSystem, path, file);
 
+	if (iDoTheExportThing.mOutput.fail() || iDoTheExportThing.mOutput.bad()) {
```
Comment (angelowang) [inline code/ColladaExporter.cpp]: I tried and `bad()` will also lead to tellp()==1.
So perhaps we should use !.good() instead.
There is a table in this page seems the only exception is eofbit will also make good() returning false.
http://en.cppreference.com/w/cpp/io/basic_ios/fail

**Comment by angelowang** [comment]
@christinelehmann

**Comment by larsxschneider** [comment]
Please don't use this repo anymore. Please use this: https://git.autodesk.com/autodesk-forks/assimp (both of you should have write access ... see email a few minutes ago)

**Comment by angelowang** [comment]
This PR was sent right before your email:) I will do it in autodesk-forks.

File: code/ColladaExporter.cpp (hunk 1 above)
Comment (ndotl) [inline code/ColladaExporter.cpp]: This is Haik. from the table (and fail() impl), fail() will always return true if bad() returns true, so it's ok to only check fail() (tellp() does the same thing).

Comment (angelowang) [inline code/ColladaExporter.cpp]: It's cool to see all different alias, like `ztou` is for Johnny :)

**Pull request closed without merging**